fix: add feature flag for ruzstd deps

Xynnn007 opened the pull request:
When we disable feature `zstd`, the compiler will cry with the following information

```
12 | use ruzstd::decoding::errors::FrameDecoderError;
   |     ^^^^^^ use of undeclared crate or module `ruzstd`
```

This is because the ruzstd is disabled but the code still imports it. This patch fixes this by add a feature flag.

klhowell reviewed (commented):
Thanks for this patch. In addition to your stated change, it also makes several other lint changes that I don't want to merge as part of this single commit. If you can re-submit as a clean patch with only the #[cfg] fix then I'll happily merge it.

Xynnn007 committed:
fix: add feature flag for ruzstd deps

When we disable feature zstd, the compiler will cry with the following
information

```
12 | use ruzstd::decoding::errors::FrameDecoderError;
   | ^^^^^^ use of undeclared crate or module ruzstd
```

This is because the ruzstd is disabled but the code still imports it.
This patch fixes this by add a feature flag.

Signed-off-by: Xynnn007 <xynnn@linux.alibaba.com>

Xynnn007 commented:
@klhowell Thanks for the response. Updated.
